blocks/log.py

```diff
@@ -22,6 +60,279 @@ class TrainingLog(defaultdict):
         By default it contains ``iterations_done``, ``epochs_done`` and
         ``_epoch_ends`` (a list of time stamps when epochs ended).
 
+    """
+    def __init__(self, uuid=None):
+        if uuid is None:
+            self.uuid = uuid4()
+        else:
+            self.uuid = uuid
+        if uuid is None:
+            self.status.update({
```

Comment:
By the child class. Basically, the child class constructs an empty status object first, which adheres to the `MutableMapping` interface but stores its data in e.g. a database table, or a dictionary. The parent `_TrainingLog` is then responsible for adding the first default values to this `self.status` object.